drivers/hmis/app/graphql/mutations/update_bed_nights.rb

```diff
@@ -0,0 +1,46 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class UpdateBedNights < BaseMutation
+    argument :enrollment_ids, [ID], required: true
+    argument :bed_night_date, GraphQL::Types::ISO8601Date, required: true
+    argument :action, Types::HmisSchema::Enums::BulkActionType, required: true
+
+    field :success, Boolean, null: true
+
+    def resolve(enrollment_ids:, bed_night_date:, action:)
+      enrollments = Hmis::Hud::Enrollment.viewable_by(current_user).where(id: enrollment_ids)
+      raise 'not found' unless enrollments.count == enrollment_ids.uniq.length
+      raise 'project mismatch' unless enrollments.map(&:project_id).uniq.length == 1
+
+      hud_user_id = Hmis::Hud::User.from_user(current_user).user_id
+
+      Hmis::Hud::Service.transaction do
+        case action
+        when 'ADD'
+          services = enrollments.map do |enrollment|
+            Hmis::Hud::Service.new(
+              **enrollment.slice(:enrollment_id, :personal_id, :data_source_id),
+              date_provided: bed_night_date,
+              record_type: 200, # bed night
+              type_provided: 200, # bed night
```

Comment:
We have the generic "bulk services" feature which is currently hidden in the frontend, because its not MVP ready. At some point yeah I we need to either integrate the two, or get the bulk services up to par. The main difference with the generic tool is that it can handle any arbitrary fields (eg financial amount) and so submission should occur using the form processor, not a custom mutation.

Reply:
Doing `HudUtility2024.record_type('Bed Night')` seems almost worse to me, because that's a display string - we should be free to change it without impacting code. I agree with getting numbers out of the code, but I think we need some kind of enums / constants to do that.

Reply:
I'm behind using some enums or constants.  Just don't think we have them defined yet.

Reply:
Lets figure something out for hud utility but it's out of scope for this PR. I can do something on my hmis 2024 branch